### PR TITLE
feature: add support for Std:Join for pipelines

### PR DESCRIPTION
**Pull request by metrizable**
*Issue #, if available:*

*Description of changes:*

Changes include:
* Support for SageMaker Pipelines DSL function `Std:Join`

*Testing done:*

unit:

```
❯ tox --parallel 3 -- -rfE --disable-warnings tests/unit
✔ OK black-format in 14.852 seconds
✔ OK flake8 in 33.196 seconds
✔ OK pylint in 45.187 seconds
✔ OK docstyle in 31.303 seconds
✔ OK twine in 9.889 seconds
✔ OK doc8 in 13.097 seconds
✔ OK sphinx in 56.37 seconds
✔ OK py36 in 6 minutes, 7.25 seconds
✔ OK py38 in 5 minutes, 38.155 seconds
✔ OK py37 in 6 minutes, 14.6 seconds
```

integ:

```
❯ IGNORE_COVERAGE="#" tox -e py38 -- -rfE --no-header --disable-warnings --cov-report= -s tests/integ/test_workflow.py::test_three_step_definition
============================================================== test session starts ==============================================================
collected 1 item

tests/integ/test_workflow.py .



============================================================== 1 passed in 10.56s ===============================================================
____________________________________________________________________ summary ____________________________________________________________________
  py38: commands succeeded
  congratulations :)
  12.40s user 0.76s system 77% cpu 17.027 total
```

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/unit/sagemaker/workflow/test_functions.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (metrizable) [inline tests/unit/sagemaker/workflow/test_functions.py:1]: Should be 2021
```suggestion
# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

**Review by owen-t — DISMISSED** [review]
- function
- integ test
- unit test

LGTM

File: tests/integ/test_workflow.py [hunk 2]
```diff
@@ -154,7 +150,20 @@ def test_three_step_definition(
         ],
         outputs=[
             ProcessingOutput(output_name="train_data", source="/opt/ml/processing/train"),
-            ProcessingOutput(output_name="test_data", source="/opt/ml/processing/test"),
+            ProcessingOutput(
+                output_name="test_data",
+                source="/opt/ml/processing/test",
+                destination=Join(
```
Comment (navaj0) [inline tests/integ/test_workflow.py:156]: Nit. Is it possible to make create a default destination with this pattern?

**Review by navaj0 — APPROVED** [review]

File: tests/integ/test_workflow.py (hunk 2 above)
Comment (metrizable) [reply tests/integ/test_workflow.py:156]: default destination already supported by sdk

**Review by ahsan-z-khan — APPROVED** [review]

**Pull request merged**